react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+                {
+                    jsi::Object obj = prop.getObject(*runtime);
+                    prop = obj.getProperty(*runtime, p.toRawUTF8());
+                }
+            }
+
+            auto func = prop.getObject(*runtime).asFunction(*runtime);
+
+            std::vector<jsi::Value> jsiArgs;
+            for (auto &v : vargs)
+            {
+                jsiArgs.push_back(varToJSIValue(v, *runtime));
+            }
+
+            const jsi::Value *argsPtr = jsiArgs.data();
+            jsi::Value result = func.call(*runtime, argsPtr, jsiArgs.size());
+
+            return jsiValueToVar(result, *runtime);
+        }
+
+        //==============================================================================
+        jsi::Function createSetTimerFunction(bool isInterval)
+        {
+            return jsi::Function::createFromHostFunction(
+                *runtime,
+                jsi::PropNameID::forAscii(*runtime, ""),
+                0,
+                [=] (jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)
+                {
+                    juce::ignoreUnused(thisVal);
+
+                    jsi::Function fn      = args[0].asObject(rt).asFunction(rt);
+                    const auto    timeout = static_cast<int>(args[1].asNumber());
+
+                    std::vector<jsi::Value> timeoutArgs;
+                    if (count > 2)
+                    {
+                        for (size_t i = 2; i < count; ++i)
+                        {
+                            timeoutArgs.emplace_back(jsi::Value(rt, args[i]));
+                        }
+                    }
+
+                    return timeoutsManager->newTimeout(std::move(fn), timeout, std::move(timeoutArgs), isInterval);
+                }
+            );
+        }
+
+        jsi::Function createClearTimerFunction()
+        {
+            return jsi::Function::createFromHostFunction(
+                *runtime,
+                jsi::PropNameID::forAscii(*runtime, ""),
+                0,
+                [this] (jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)
+                {
+                    juce::ignoreUnused(rt);
+                    juce::ignoreUnused(thisVal);
+                    juce::ignoreUnused(count);
+
+                    const auto timerId = static_cast<int>(args[0].asNumber());
+                    return timeoutsManager->clearTimeout(timerId);
+                }
+            );
+        }
+
+        //==============================================================================
+        void reset()
+        {
+            if (timeoutsManager)
+                timeoutsManager->clear();
+
+            runtime         = facebook::hermes::makeHermesRuntime();
+            timeoutsManager = std::make_unique<TimeoutFunctionManager>(*runtime);
+
+            // Quick and dirty console object provide. Could be improved upon.
+            jsi::Function logFunction =
```

Comment:
@nick-thompson , Curious to hear if you think this is a sufficient enough implementation for `console.log()` bindings for now?

Reply:
Yep, I think so. I think in the long run we will want to make the `console.log` configurable at the EcmascriptEngine interface so people can slice in whatever logging infrastructure they want, but for now this is fine